Yubico.YubiKey/docs/users-manual/sdk-programming-guide/secure-channel-protocol.md

```diff
@@ -16,492 +16,549 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+IReadOnlyCollection<X509Certificate2> certificateList;
+using (var session = new SecurityDomainSession(yubiKeyDevice))
+{
+    certificateList = session.GetCertificates(keyReference);
+}
+
+// Create parameters using leaf certificate
+var leaf = certificateList.Last();
+var ecDsaPublicKey = leaf.PublicKey.GetECDsaPublicKey()!.ExportParameters(false);
+var keyParams = new Scp11KeyParameters(
+    keyReference, 
+    new ECPublicKeyParameters(ecDsaPublicKey));
+
+// Use with any application
+using var pivSession = new PivSession(yubiKeyDevice, keyParams);
 ```
 
-```txt
-   slot 1:   --empty--
-   slot 2:   --empty--
-   slot 3: KVN=3  ENC       MAC       DEK
-```
+### SCP11a Example
 
-Now it is possible to delete the KVN=3 key set using an `Scp03Session` created using
-the KVN=3 key set. Pass true for the second argument.
+Full mutual authentication requires more setup:
 
 ```csharp
-    using StaticKeys scp03Keys = GetStaticKeys(3);
-    using (var scp03Session = new Scp03Session(yubiKeyDevice, scp03Keys))
-    {
-        scp03Session.DeleteKeySet(3, true);
-    }
+// Start with default SCP03 connection
+using var session = new SecurityDomainSession(yubiKeyDevice, Scp03KeyParameters.DefaultKey);
+
+const byte kvn = 0x03;
+var keyRef = new KeyReference(ScpKeyIds.Scp11A, kvn);
+
+// Generate new key pair on YubiKey
+var newPublicKey = session.GenerateEcKey(keyRef, 0);
+
+// Setup off-card entity (OCE)
+var oceRef = new KeyReference(OceKid, kvn);
+var ocePublicKey = new ECPublicKeyParameters(oceCerts.Ca.PublicKey.GetECDsaPublicKey()!);
+session.PutKey(oceRef, ocePublicKey, 0);
+
+// Store CA identifier
+var ski = GetSubjectKeyIdentifier(oceCerts.Ca);
+session.StoreCaIssuer(oceRef, ski);
+
+// Create SCP11a parameters
+var scp11Params = new Scp11KeyParameters(
+    keyRef,
+    new ECPublicKeyParameters(newPublicKey.Parameters),
+    oceRef,
+    new ECPrivateKeyParameters(privateKey),
+    certChain);
+
+// Use the secure connection
+using var session = new SecurityDomainSession(yubiKeyDevice, scp11Params);
 ```
 
-Passing `true` lets the SDK know this is the last key to be deleted. It can then call the
-DELETE KEY command with the appropriate parameters, and the YubiKey will be able to delete
-the key set. After this delete, the state of SCP03 key sets will be the following:
+### Security Considerations
 
-```txt
-   slot 1: KVN=0xff  KeyId=1:ENC(default)  KeyId=2:MAC(default)  KeyId=3:DEK(default)
-   slot 2:   --empty--
-   slot 3:   --empty--
+1. **Certificate Management:**
+   - Proper certificate validation is crucial
+   - Consider using certificate allowlists
+   - Manage certificate chains carefully
+
+2. **Key Generation:**
+   - Can generate keys on YubiKey or import existing
+   - YubiKey-generated keys never leave the device
+   - Imported keys must be properly protected
+
+3. **Protocol Selection:**
+   - SCP11b: Simpler but less secure
+   - SCP11a: Better security through mutual auth
```

Comment:
I don't think it's really appropriate to say it'r "more" or "less" secure. They offer different types of authentication and it depends on what your needs are.